var/spack/repos/builtin/packages/dmlc-core/package.py

```diff
@@ -0,0 +1,46 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class DmlcCore(CMakePackage):
+    """DMLC-Core is the backbone library to support all DMLC projects,
+    offers the bricks to build efficient and scalable
+    distributed machine learning libraries."""
+
+    homepage = "https://github.com/dmlc/dmlc-core"
+    url      = "https://github.com/dmlc/dmlc-core.git"
+
+    version('master', git='https://github.com/dmlc/dmlc-core.git')
+    version('a6c5701', git='https://github.com/dmlc/dmlc-core.git',
```

Comment:
This version won't sort very well as soon as you add another version. It's usually better to name the version after the date of the commit, like `2017-5-8`.